packages/mui-docs/tsconfig.build.json

```diff
@@ -3,14 +3,23 @@
   // Actual .ts source files are transpiled via babel
   "extends": "./tsconfig.json",
   "compilerOptions": {
+    "paths": {
```

Comment:
I suppose the cleanest solution would be to use the build output of icons-material, but that would yield worse DX.

One other way it could work is to create a single .d.ts file (say, Icon.d.ts) in icons-material with
```ts
export { default } from '@mui/material/SvgIcon';
```

Then to change the paths in tsconfig.json to `"@mui/icons-material/*": ["./packages/mui-icons-material/src/Icon.d.ts"],`, so there's a single declaration file covering all the icons. In the future, when we have the exports field in package json, we could even have `"./*": { "types": "./src/Icon.d.ts", ... }` and get rid of generating a separate .d.ts per icon (but it's just a random thought, I haven't tested it).